webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -202,7 +203,7 @@ public void writeStatusAndHeaders(Http.ResponseStatus status, Map<String, List<S
         boolean lengthSet = HttpUtil.isContentLengthSet(response);
         forcedChunked = HttpUtil.isTransferEncodingChunked(response);
         if (!lengthSet) {
-            lengthOptimization = status.code() == Http.Status.OK_200.code() && !forcedChunked && !isSseEventStream(headers);
+            lengthOptimization = isSseEventStream(headers) ? false : lengthOptimization;
```

Comment:
lengthOptimization = !isSseEventStream(headers)

Reply:
Or is it that we can do `lengthOptimization = !isSseEventStream(headers) && !forcedChunk` and make `forcedChunk` just a local variable instead of an instance variable. Seems like there's only one test for it here `lengthOptimization && !forcedChunked`, but I'm not 100% sure.

Reply:
Elimintarted global variable `forcedChunked`. Previously, it exists to check if user added transfer-encoding-chunked into his response handler and only required when lengthOptimization is being determined. Because we set the response to transfer-encoding-chunked at the start of response processing if there is no content-length, hence why that state is needed via that global variable when we finally checked if lengthOptimization will take place. The new change removes the setting of the response to  transfer-encoding-chunked when length does not exist at the start (i.e. in `writeStatusAndHeaders`) and moves it to a part when header is set in onNextPipe where we know lengthOptimization is not required anymore. This way we can make sure that at the end in writeLastContent() that we can check if the response was forced chunked via the HttpUtil.isTransferEncodingChunked when we determine if lengthOptimization is needed.